ldclient_migration_test.go

```diff
@@ -1,61 +1,97 @@
 package ldclient
 
 import (
+	gocontext "context"
 	"testing"
 
+	"github.com/launchdarkly/go-sdk-common/v3/ldcontext"
 	"github.com/launchdarkly/go-sdk-common/v3/ldmigration"
 	"github.com/launchdarkly/go-sdk-common/v3/lduser"
 	"github.com/launchdarkly/go-sdk-common/v3/ldvalue"
 	"github.com/launchdarkly/go-server-sdk-evaluation/v3/ldbuilders"
+	"github.com/launchdarkly/go-server-sdk/v7/interfaces"
 	"github.com/stretchr/testify/assert"
 )
 
 var migrationTestUser = lduser.NewUser("userkey")
 
-func TestDefaultIsReturnedIfFlagEvaluatesToNonStringType(t *testing.T) {
-	flag := ldbuilders.NewFlagBuilder("migration-key").Build() // flag is off and we haven't defined an off variation
+type MigrationVariationMethod = func(
+	client *LDClient,
+	key string,
+	context ldcontext.Context,
+	stage ldmigration.Stage,
+) (ldmigration.Stage, interfaces.LDMigrationOpTracker, error)
+
+func TestMigrationVariation(t *testing.T) {
```

Comment:
Re-worked these tests to be parameterized so they could be ran against `MigrationVariation` and `MigrationVariationEx`.